src/deluge/model/global_effectable/global_effectable_for_clip.cpp

```diff
@@ -114,7 +112,7 @@ void GlobalEffectableForClip::renderOutput(ModelStackWithTimelineCounter* modelS
 	static StereoSample globalEffectableBuffer[SSI_TX_BUFFER_NUM_SAMPLES] __attribute__((aligned(CACHE_LINE_SIZE)));
 
 	bool canRenderDirectlyIntoSongBuffer =
-	    !isKit() && !filterSetConfig.doLPF && !filterSetConfig.doHPF && !delayWorkingState.doDelay
+	    !isKit() && filterSets[0].isLPFOn() && !filterSets[1].isHPFOn() && !delayWorkingState.doDelay
```

Comment:
missing `!` for `isLPFOn`? (see my comment above about combining these into 1 method too)